analyzers/src/SonarAnalyzer.CSharp/Rules/CloudNative/AzureFunctionsLogFailures.cs

```diff
@@ -0,0 +1,168 @@
+                            }
+                        }
+                    },
+                    SyntaxKind.CatchClause);
+                }
+            });
+
+        private static bool HasLoggerInScope(SemanticModel semanticModel, int position, IMethodSymbol entryPoint, CancellationToken cancellationToken) =>
+            entryPoint.Parameters.Any(x => x.Type.DerivesOrImplements(KnownType.Microsoft_Extensions_Logging_ILogger))
+                // Instance method entry points might have access to an ILogger via injected fields/properties
+                // https://docs.microsoft.com/en-us/azure/azure-functions/functions-dotnet-dependency-injection
+                || (entryPoint is { IsStatic: false, ContainingType: { } container }
+                    && container.AllAccessibleMembers(semanticModel, position, cancellationToken)
+                        .Any(x => x.GetSymbolType()?.DerivesOrImplements(KnownType.Microsoft_Extensions_Logging_ILogger) is true));
+
+        private sealed class LoggerCallWalker : SafeCSharpSyntaxWalker
+        {
+            private readonly SemanticModel model;
+            private readonly ITypeSymbol iLogger;
+            private readonly CancellationToken cancellationToken;
+            private List<Location> invalidInvocations;
+
+            public LoggerCallWalker(SemanticModel model, ITypeSymbol iLoggerSymbol, CancellationToken cancellationToken)
+            {
+                this.model = model;
+                iLogger = iLoggerSymbol;
+                this.cancellationToken = cancellationToken;
+            }
+
+            public bool HasValidLoggerCall { get; private set; }
+            public IEnumerable<Location> InvalidLoggerInvocationLocations => invalidInvocations;
+
+            public override void Visit(SyntaxNode node)
+            {
+                if (!HasValidLoggerCall)
+                {
+                    cancellationToken.ThrowIfCancellationRequested();
+                    base.Visit(node);
+                }
+            }
+
+            public override void VisitInvocationExpression(InvocationExpressionSyntax node)
+            {
+                if (model.GetSymbolInfo(node, cancellationToken).Symbol is IMethodSymbol { ReceiverType: { } receiver } methodSymbol
+                    && receiver.DerivesOrImplements(iLogger))
+                {
+                    if (IsValidLogCall(node, methodSymbol))
+                    {
+                        HasValidLoggerCall = true;
+                    }
+                    else
+                    {
+                        invalidInvocations ??= new();
+                        invalidInvocations.Add(node.GetLocation());
+                    }
+                }
+                base.VisitInvocationExpression(node);
+            }
+
+            public override void VisitArgument(ArgumentSyntax node)
```

Comment:
`VisitArgument` makes sense for static azure functions. For Azure Functions with DI it doesn't make sense, because there is no need to pass the logger around. We should consider removing this method or only running it in a static azure function.